### PR TITLE
Acutal HTTP request making infrastructure

### PR DESCRIPTION
**Pull request by rsjethani**
The new `request()` function is what makes the actual HTTP transaction. The higher level APIs simply prepare the request based on resource type, HTTP method etc. and then call this function to do the real job.

### CODE REVIEW

**Comment by rsjethani** [comment]
cc @chb-ioki @StefMa

File: request.go
Comment (StefMa) [inline request.go:1]: Same as https://github.com/ioki-mobility/go-outline/pull/7#discussion_r1249684381 🙃

Comment (rsjethani) [reply request.go:1]: @StefMa  all code in this PS is not exposed outside outline package but sure lets talk to avoid all confusion

**Review by StefMa — DISMISSED** [review]
Clarified in a call.
:shipit:

**Review by StefMa — APPROVED** [review]
LGTM 👌

**Pull request merged**